src/code-generation/CodeGeneratorBase.ts

```diff
@@ -98,14 +98,27 @@ abstract class CodeGeneratorBase<TSettings extends CodeGeneratorBaseSettings> im
     --this._branches.length
     return this
   }
+  wackIndent(oneLine: boolean): this {
```

Comment:
The changes in this file has broken code generation for other languages

Reply:
What should I do to fix this? I came up with some weird solution because I was unsure of a better way.

Reply:
You can override the methods in your Lua-specific class if you need specific functionality. But some of these things could probably be generalized. I'll add some comments that could help you fix it.